### PR TITLE
Add thing docker and cli instructions.

### PR DESCRIPTION
**Pull request by vitbaq**
On this PR two new sections are included on knot documentation.
The first one is instructions for KNoT CLI and second is instructions for Zephyr KNoT SDK Docker environment.

### CODE REVIEW

File: doc/thing/thing-cli.rst [hunk 1]
```diff
@@ -0,0 +1,92 @@
+Using CLI
+=========
+
```
[the remark this answers is not in the document]
Comment (vitbaq) [reply doc/thing/thing-cli.rst]: Ok, I will include a requirements page.

File: doc/thing/thing-requirements.rst [hunk 2]
```diff
@@ -0,0 +1,147 @@
+Requirements
+============
+
+Linux based OS
+--------------
+
+The actual version of KNoT Zephyr SDK is only available for Linux based systems.
+You may consider using a Virtual Machine running a Linux distribution on other systems.
```
Comment (jcorde) [inline doc/thing/thing-requirements.rst]: Please, also suggest the usage of Docker for it. We should use Docker prior than use VMs

File: doc/thing/thing-requirements.rst [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+- Program the ``$HOME/.profile`` to always source zephyr-env.sh when you log in.
+
+.. code-block:: bash
+
+   $ echo "source $ZEPHYR_BASE/zephyr-env.sh" >> $HOME/.profile
+
+.. note:: If you skip this step, it will be necessary to manually source zephyr-env.sh every time a new terminal is opened.
+
+
+nRF5x Command Line Tools
```
Comment (jcorde) [inline doc/thing/thing-requirements.rst]: You might merge the `nRF5x Command Line Tools` and `Install Segger JLink` sections by telling the user to download [the nRF Command Line Tools package](https://www.nordicsemi.com/-/media/Software-and-other-downloads/Desktop-software/nRF-command-line-tools/sw/Versions-10-x-x/nRFCommandLineTools1021Linuxamd64tar.gz) and install the `JLink_Linux_V644e_x86_64.deb` and `nRF-Command-Line-Tools_10_2_1_Linux-amd64.deb` files.

You can find the commands between https://github.com/CESARBR/zephyr-knot-sdk/blob/master/docker/Dockerfile#L126 and https://github.com/CESARBR/zephyr-knot-sdk/blob/master/docker/Dockerfile#L138

File: doc/thing/thing-requirements.rst (hunk 2 above)
Comment (vitbaq) [reply doc/thing/thing-requirements.rst]: Done

File: doc/thing/thing-requirements.rst (hunk 3 above)
Comment (vitbaq) [reply doc/thing/thing-requirements.rst]: Done

**Pull request merged**